PROJECT-FLOW.md

```diff
@@ -0,0 +1,60 @@
+# Project flow
+
+## How the voting occurs
+
+1. A person will create a group and can add students to it / invite the students
+2. Every participant of the group will be awarded a set number of tokens when they join the group. This token count is encrypted with the user's master key and stored in the database. (We'll get to where it is stored later)
+3. To create a **poll**, the admin can create a form with a question along with different voting options( can be a MCQ or a Y/N ).
+
+> can be done by displaying a form where the admin can create a question and some option, along with the polling date and time.
+
+4. During the poll, the participant can pick his desired option quadratically.
+5. His/ her vote is encrypted and sent to the server along with the zn-proofs verifying his vote satisfies the requirements, which are
+
+- The vote should be a part of one of the options given in the poll.
+- The person has enough tokens to cast his/her vote (ex: the person needs to have 9 tokens to purchase 3 votes)
+
+6. This data is then stored in the server
+7. After the polling, the results are calculated by secure Multi-Party-Computation (MPC), this is done by splitting the votes between all the voters and computing the result. _( not sure how we verify if the votes are valid during the MPC )._ **Problem: All the voters need to be online to perform MPC.**
+8. The result is computed and announced. This is saved in the database in encrypted form.
+
+## Schemas
+
+### 1. User Schema
+
+- Username
+- Email id _( preferably webmail, for proving one-person one-account)_
+- Password
+- Array of objects containing
+  - the name of the group the user is part of
+  - Number of tokens he/she has left in it._(Encrypted)_
```

Comment:
How will you be representing this array in MySQL tables?

Reply:
The same for all arrays mentioned below.

Reply:
By creating some relationship tables.

  #### 1. a. Token Relationship Table

  - User Id
  - Group Id
  - Token's left _( Encrypted with the user's master key )_

  #### 3.a Vote Relationship Table

  - Voter Id
  - Poll id
  - Encrypted Vote

Wasn't sure if I should add them in Schema as these technically aren't models

Reply:
The schema should consist of the SQL tables and fields. One should be able to create the db by going through the schema i.e., there should be no ambiguity of how a field is to be implemented. Define the SQL data types as well for each field. This will save a lot of work later.